jfrog-cli/jfrog/buildinfo_test.go

```diff
@@ -38,6 +41,85 @@ func TestBuildAddDependenciesFromHomeDir(t *testing.T) {
 	cleanArtifactoryTest()
 }
 
+func TestBuildPromote(t *testing.T) {
+	initArtifactoryTest(t)
+	buildName, buildNumberA := "cli-test-build", "10"
+
+	// Upload files with buildName and buildNumber
+	specFileA, err := tests.CreateSpec(tests.SplitUploadSpecA)
+	if err != nil {
+		t.Error(err)
+	}
+	artifactoryCli.Exec("upload", "--spec="+specFileA, "--build-name="+buildName, "--build-number="+buildNumberA)
+	artifactoryCli.Exec("build-publish", buildName, buildNumberA)
+
+	key1 := "key"
+	value1 := "v1,v2"
+	key2 := "another"
+	value2 := "property"
+	artifactoryCli.Exec("build-promote", buildName, buildNumberA, tests.Repo2, fmt.Sprintf("--props=%s=%s;%s=%s", key1, value1, key2, value2))
+	buildInfo := inttestutils.GetBuildInfo(artifactoryDetails.Url, buildName, buildNumberA, t, artHttpDetails)
+	resultItems := getResultItemsFromArtifactory(tests.SearchRepo2, t)
+
+	if len(buildInfo.Modules[0].Artifacts) != len(resultItems) {
+		t.Error("Incorrect number of artifacts were uploaded, expected:", len(buildInfo.Modules[0].Artifacts), " Found:", len(resultItems))
+	}
+
+	propsMap := make(map[string]string)
+	propsMap["build.name"] = buildInfo.Name
+	propsMap["build.number"] = buildInfo.Number
+	propsMap[key1] = value1
+	propsMap[key2] = value2
+
+	validateArtifactsProperties(resultItems, t, propsMap)
+	cleanArtifactoryTest()
+}
+
+// Returns the artifacts result found by the provided spec
+func getResultItemsFromArtifactory(specName string, t *testing.T) []rtutils.ResultItem {
+	searchGoSpecFile, err := tests.CreateSpec(specName)
+	if err != nil {
+		t.Error(err)
+	}
+	spec, flags := getSpecAndCommonFlags(searchGoSpecFile)
+	flags.SetArtifactoryDetails(artAuth)
+	var resultItems []rtutils.ResultItem
+	for i := 0; i < len(spec.Files); i++ {
+		searchParams, err := generic.GetSearchParams(spec.Get(i))
+		if err != nil {
+			t.Error(err)
+		}
+
+		currentResultItems, err := services.SearchBySpecFiles(searchParams, flags, rtutils.ALL)
+		if err != nil {
+			t.Error("Failed Searching files:", err)
+		}
+		resultItems = append(resultItems, currentResultItems...)
+	}
+	return resultItems
+}
+
+// This function validates the properties on the provided artifacts. Every property within the provided map should appear on the artifact.:
+func validateArtifactsProperties(resultItems []rtutils.ResultItem, t *testing.T, propsMap map[string]string) {
+	for _, item := range resultItems {
+		properties := item.Properties
+		if len(properties) < 1 {
+			t.Error("Failed finding properties on item:", item.GetItemRelativePath())
+		}
+		propertiesMap := convertSliceToMap(properties)
```

Comment:
The *convertSliceToMap* function is currently part of the *buildtools_test.go*, but is now also used by this file. I therefore suggest moving it to *utils/tests/utils.go*.

Reply:
Done